### PR TITLE
Diann update 

### PR DESCRIPTION
**Pull request by ypriverol**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/quantms/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/quantms _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/assemble_empirical_library/meta.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
-name: diannsearch
-description: A module for DIA library free analysis based on DIA-NN.
+name: assemble_empirical_library
+description: A module for assemble empirical library analysis based on DIA-NN.
```
Comment (jpfeuffer) [inline modules/local/assemble_empirical_library/meta.yml:2]:
```suggestion
description: A module for assembling an empirical transition library based on a preliminary analysis of the in-silico library with DIA-NN.
```

File: modules/local/assemble_empirical_library/meta.yml [hunk 2]
```diff
@@ -11,34 +10,30 @@ tools:
       homepage: https://github.com/vdemichev/DiaNN
       documentation: https://github.com/vdemichev/DiaNN
 input:
+  - mzMLs:
+      type: file
+      description: Spectra file in mzML format
+      pattern: "*.mzML"
+  - quant:
+      type: file
+      description: The .quant files containing IDs and quantification information
```
Comment (jpfeuffer) [inline modules/local/assemble_empirical_library/meta.yml:19]:
```suggestion
      description: The .quant files from DIA-NN preliminary analysis, containing IDs and quantification information.
```

File: modules/local/diann_preliminary_analysis/meta.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: diann_preliminary_analysis
+description: A module for preliminary analysis of individual raw files based on DIA-NN.
```
Comment (jpfeuffer) [inline modules/local/diann_preliminary_analysis/meta.yml:2]:
```suggestion
description: A module for preliminary analysis of individual raw files with DIA-NN using the in-silico generated library (also from DIA-NN).
```

File: modules/local/diannsummary/meta.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      description: Spectra files in mzML format.
+      pattern: "*.mzML"
+  - fasta:
+      type: file
+      description: Protein sequence database in Fasta format.
+      pattern: "*.{fasta,fa}"
+  - quant:
+      type: file
+      description: Identification and Quantification file from DIA-NN.
+      pattern: "*.{fasta,fa}"
```
Comment (jpfeuffer) [inline modules/local/diannsummary/meta.yml:28]:
```suggestion
      pattern: "*.quant"
```

File: modules/local/diannsummary/meta.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+name: diannsummary
+description: A module for summarization the info from DIA-NN analysis.
```
Comment (jpfeuffer) [inline modules/local/diannsummary/meta.yml:2]:
```suggestion
description: A module for summarization of results from DIA-NN analysis.
```

File: modules/local/individual_final_analysis/meta.yml [hunk 6]
```diff
@@ -11,35 +10,38 @@ tools:
...
-      pattern: "*.mzML"
-  - fasta:
+      description: DIA-NN log file
+      pattern: "*.log"
+  - library:
       type: file
-      description: FASTA sequence databases
-      pattern: "*.{fasta,fa}"
-  - cfg:
+      description: Silico-predicted spectral library by deep leaning predictor in DIA-NN
```
Comment (jpfeuffer) [inline modules/local/individual_final_analysis/meta.yml:22]: Isn't this using the empirical library?? Not the initially predicted one

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
Also, changelog missing. And documentation updates.

File: bin/msstats_tmt.R [hunk 7]
```diff
@@ -246,5 +241,28 @@ if (l == 1) {
...
+    valid_comp_data <- test.MSstatsTMT$ComparisonResult[!is.na(test.MSstatsTMT$ComparisonResult$pvalue), ]
+
+    if (nrow(valid_comp_data[!duplicated(valid_comp_data$Protein),]) < 2) {
+        warning("Warning: Not enough proteins with valid p-values for comparison. Skipping groupComparisonPlots step!")
+    } else {
+        require(MSstats)
+        # BUG groupComparisonPlots function: re-run OpenMStoMSstatsTMTFormat
+        quant <- OpenMStoMSstatsTMTFormat(data, useUniquePeptide=useUniquePeptide, rmPSM_withfewMea_withinRun=rmPSM_withfewMea_withinRun,
+            rmProtein_with1Feature=rmProtein_with1Feature
+        )
```
[the remark this answers is not in the document]
Comment (jpfeuffer) [reply bin/msstats_tmt.R:255]: If MSstats is needed for those plots, it needs to go into the dependencies!

[the remark this answers is not in the document]
Comment (ypriverol) [reply bin/msstats_tmt.R:255]: @daichengxin can you take care of this comment. Please do your changes in the diann branch.

File: bin/msstats_tmt.R [hunk 8]
```diff
@@ -246,5 +241,28 @@ if (l == 1) {
     test.MSstatsTMT <- groupComparisonTMT(contrast.matrix=contrast_mat, data=processed.quant)
```
[the remark this answers is not in the document]
Comment (ypriverol) [reply bin/msstats_tmt.R:241]: @daichengxin can you take care of this comment. Please do your changes in the diann branch.

**Comment by ypriverol** [comment]
LFQ is failing @daichengxin

**Comment by jpfeuffer** [comment]
Weird, seems to work now. Conda hiccups are normal though.

**Comment by ypriverol** [comment]
@jpfeuffer Can I merge now and convert your comments into issue ;)

**Comment by jpfeuffer** [comment]
Yes:
- MSstats TMT weird code and import of MSstats
- Changelog and docu
- double check usage of empirical vs insilico library in module descriptions

**Comment by ypriverol** [comment]
#183 issue created by @jpfeuffer

**Pull request merged**